src/is/L42/flyweight/P.java

```diff
@@ -103,10 +117,26 @@ public boolean hasUniqueNum() {
       return false;

```

Comment:
should hasUniqueNum become cached and precomputed during the construction? probably not, but you are in the right position to test it.